test/e2e/common/node/runtimeclass.go

```diff
@@ -51,7 +51,21 @@ var _ = SIGDescribe("RuntimeClass", func() {
 		handler := f.Namespace.Name + "-handler"
 		rcName := createRuntimeClass(f, "unconfigured-handler", handler)
 		pod := f.PodClient().Create(e2enode.NewRuntimeClassPod(rcName))
-		expectSandboxFailureEvent(f, pod, handler)
+		eventSelector := fields.Set{
+			"involvedObject.kind":      "Pod",
+			"involvedObject.name":      pod.Name,
+			"involvedObject.namespace": f.Namespace.Name,
+			"reason":                   events.FailedCreatePodSandBox,
+		}.AsSelector().String()
+		// Events are unreliable, don't depend on the event. It's used only to speed up the test.
+		err := e2eevents.WaitTimeoutForEvent(f.ClientSet, f.Namespace.Name, eventSelector, handler, framework.PodEventTimeout)
+		if err != nil {
+			framework.Logf("Warning: did not get event about FailedCreatePodSandBox")
+		}
+		// Check the pod is still not running
+		p, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "could not re-read the pod after event (or timeout)")
+		framework.ExpectEqual(p.Status.Phase, v1.PodPending, "Pod phase isn't pending")
```

Comment:
anything we can check on the pod? with this change any slow scheduling pod (more than 2 minutes) will pass. Ideally some validation of the fact that pod cannot run is needed.

Reply:
Hi Sergey, I believe there is no good way to validate the pod can't run. I copied this pattern from other e2e tests that use `WaitTimeoutForEvent`, e. g. [here](https://github.com/kubernetes/kubernetes/blob/3f98b98c6944d0d2a9c945ffe547103151c0619f/test/e2e/storage/testsuites/volumemode.go#L243). There is no validation a pod can't run in any of them.

We could increase the timeout to reduce the likelihood slow scheduling leads to a passing test that should fail. However, 2 min feels like a reasonable value to me.

Another possibility might be to check kubelet logs. I am not sure if there is a clean way to implement this check. Also, I feel asserting on logs violates the black-box nature of e2e test.

What do you think?

Reply:
/lgtm